doc/source/_templates/autosummary/class.rst

```diff
@@ -0,0 +1,33 @@
+{{ fullname | escape | underline}}
+
+.. currentmodule:: {{ module }}
+
+.. autoclass:: {{ objname }}
+
+   {% block methods %}
+   {% if methods %}
+   .. rubric:: {{ _('Methods') }}
+
+   .. autosummary::
```

Comment:
This will generate stub page for each class method and attribute which will make the doc build even longer but that's what other libraries (e.g. pandas, numpy) are doing.

Reply:
cc @maxpumperla FYI.

Reply:
@jjyao is there a compromise we could find here? this seems to more than double the build time now. I'm in touch with support from readthedocs.org right now to make longer builds happen, but for now we can't pull this off.

just asking if we can think of a trade-off and move towards a more extensive solution later?

Reply:
we can try to not autosummary class members

Reply:
yep, let's try this for now and see how it impacts build times. this PR increases rtd builds from 900s to 2000+, which will always time out

Reply:
Seems that doesn't work: https://buildkite.com/ray-project/oss-ci-build-pr/builds/11774#01863776-7701-40d6-974c-b6fe51e9d705 since we have places that manually autosummary class members which creates duplications with autoclass.

Reply:
ok, so be it. we just have to make sure we stay below 3000s total, once all other libs merge their changes. might be risky.